staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go

```diff
@@ -95,7 +95,7 @@ func (c *controller) Run(stopCh <-chan struct{}) {
 		klog.Infof("lease controller has nil lease client, will not claim or renew leases")
 		return
 	}
-	wait.Until(c.sync, c.renewInterval, stopCh)
+	wait.JitterUntil(c.sync, c.renewInterval, 0.04, true, stopCh)
```

Comment:
The alternative would be to use:
```
wait.NonSlidingUntil
```

[so that we don't move forward requests on spikes]


I don't have strong opinion either way, but we should ensure that we won't violate any assumptions:
- Kubelet can definitely tolerate this
- IIUC api-server should also be fine with this. @lavalamp - can you confirm?

Reply:
I was considering it, but preferred  to stick to sliding. Assuming that function doesn't finish within 10s we could potentially call this function again, which might cause some race-condition. I am not the expert of this part of the code, but if non-sliding is safe here I am happy to change it.

Reply:
I don't think using NonSlidingUntil will cause any race-condition. Rather, if the function takes > 10s, the next attempt will run immediately.

I think it's better to keep sliding:
* if this function took long, but eventually succeeded, then it makes sense to wait the next 10s before we refresh the lease (it's fresh when the function ends)
* if this function took long, but hasn't succeeded, it will technically make sense to retry, but I'm afraid that this may generate higher load on unhealthy master (instead of nodes refreshing statuses every 10s, we will have all nodes that are continuously inside of c.sync; the number of requests will depend on the actual backoff strategy there, but it will be still > 1qps/10s as we have with sliding).

Reply:
I took a deeper look into kube-apiserver case and it is fine - even much larger jittering would work there.

The argument about retries for non-sliding case makes sense to me. So let's keep what we have now.